tests/scripts/all.sh

```diff
@@ -822,6 +822,21 @@ component_test_platform_calloc_macro () {
     make test
 }
 
+component_test_malloc_0_null () {
+    msg "build: malloc(0) returns NULL (ASan+UBSan build)"
+    scripts/config.pl full
+    scripts/config.pl unset MBEDTLS_MEMORY_BUFFER_ALLOC_C
+    make CC=gcc CFLAGS="'-DMBEDTLS_CONFIG_FILE=\"$PWD/tests/configs/config-wrapper-malloc-0-null.h\"' -O -Werror -Wall -Wextra -fsanitize=address,undefined" LDFLAGS='-fsanitize=address,undefined'
+
+    msg "test: malloc(0) returns NULL (ASan+UBSan build)"
+    make test
+
+    msg "selftest: malloc(0) returns NULL (ASan+UBSan build)"
+    # Just the calloc selftest. "make test" ran the others as part of the
+    # test suites.
```

Comment:
How does "make test" avoid running the "calloc" selftest? I'm questioning if we need to call `make test`. I'm also questioning if we need to call calloc selftest separately if we decide we still want to run the whole `make test`

Reply:
Each test suite includes a call to the corresponding selftest function. This is why we don't bother to run `selftest` much in `all.sh`, just once to verify that the program works. This only misses the parts of `selftest.c` that aren't calling selftest functions from the various library modules.

We do need `make test` because the goal is to check that there isn't any null dereference (or similar undefined behavior) if `malloc(0)` returns null. This is very unlikely to come up in a crypto selftest. The places where it's likely to come up is in edge cases (empty input buffer, error recovery code).

Reply:
OK, so `make test` is not calling the calloc self test, because we have no test suites that call the calloc self test.